scheduler.py

```diff
@@ -2,23 +2,27 @@
 import queue
 import signal
 import sys
-import socket
+from logging_provider import logging
 
 class Scheduler:
-    def __init__(self, host, port):
+    def __init__(self):
         self.task_queue = queue.Queue()
         self.completion_status = {}
         self.lock = threading.Lock()
-        self.host = host
-        self.port = port
+        self.workers = {}
         signal.signal(signal.SIGINT, self.sigterm_handler)
-        threading.Thread(target=self.start_server, daemon=True).start()
 
     def submit_task(self, task):
         with self.lock:
             self.task_queue.put(task)
             self.completion_status[task] = False
-        print("Task submitted to scheduler:", task)
+        logging.info("Task submitted to scheduler: {}".format(task))
```

Comment:
Driver should get the future back when it submits the task right? 
Proposed soln: Have a task queue in worker, and maintain a replica in scheduler as well. Scheduler should assign the task to the worker which has least no of tasks in its local task queue. The worker should accept all tasks, create a file and return future (file name) back to scheduler and adds the task to its internal task queue. 
This wont block the driver.

Reply:
> Driver should get the future back when it submits the task right?
yes

I think the solution you mentioned can be implemented as an optimization after the more important things are done?